### PR TITLE
Nullable annotate TypeCompilationState and MessageID

### PR DESCRIPTION
**Pull request by RikkiGibson**
Bringing some annotation-only changes from my feature branch into master. /cc @jaredpar

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Errors/MessageID.cs [hunk 1]
```diff
@@ -182,6 +184,7 @@ internal enum MessageID
         IDS_FeatureNestedStackalloc = MessageBase + 12762,
         IDS_FeatureSwitchExpression = MessageBase + 12763,
         IDS_FeatureAsyncUsing = MessageBase + 12764,
+        IDS_FeatureLocalFunctionAttributes = MessageBase + 12765,
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Errors/MessageID.cs]: oops.

File: src/Compilers/CSharp/Portable/Compiler/TypeCompilationState.cs [hunk 2]
```diff
@@ -7,6 +7,8 @@
 using Microsoft.CodeAnalysis.PooledObjects;
 using Roslyn.Utilities;
 
+#nullable enable
```
Comment (sharwell) [inline src/Compilers/CSharp/Portable/Compiler/TypeCompilationState.cs]: 💡 These should be placed above the `using` directives for consistency and so developers know where to look when checking for files that do not have annotations

Comment (RikkiGibson) [reply src/Compilers/CSharp/Portable/Compiler/TypeCompilationState.cs]: Thanks, will make sure to fix.

**Comment by RikkiGibson** [comment]
@dotnet/roslyn-compiler could I please get a review on an annotation-only change?

**Pull request merged**